tests/AppBundle/Entity/Report/ReportTest.php

```diff
@@ -27,15 +28,48 @@ class ReportTest extends \PHPUnit_Framework_TestCase
 
     public function setUp()
     {
-        $this->client = m::mock(Client::class);
-        $this->report = m::mock(Report::class . '[has106Flag]', [$this->client]);
+        $this->client = m::mock(Client::class, ['getUnsubmittedReports'=>new ArrayCollection(), 'getSubmittedReports'=>new ArrayCollection()]);
+        $this->validReportCtorArgs = [$this->client, Report::TYPE_102, new \DateTime('2017-06-23'), new \DateTime('2018-06-22')];
+        $this->report = m::mock(Report::class . '[has106Flag]', $this->validReportCtorArgs);
 
         $this->gift1 = m::mock(Gift::class, ['getAmount' => 1]);
         $this->gift2 = m::mock(Gift::class, ['getAmount' => 10]);
         $this->expense1 = m::mock(Expense::class, ['getAmount' => 2]);
         $this->expense2 = m::mock(Expense::class, ['getAmount' => 20]);
     }
 
+    public static function ctorProvider()
+    {
+        return [
+            // start date, end date, submitted (true/false)
+            ['2017-06-23', '2018-06-22', [['2016-06-23', '2017-06-22', false]], 'already has unsubmitted report'],
+            ['2017-06-23', '2018-06-24', [['2016-06-23', '2017-06-22', true]], 'cannot cover more than one year'],
+            ['2017-06-24', '2018-06-23', [['2016-06-23', '2017-06-22', true], ['2015-06-23', '2016-06-22', true]], 'new report is expected to start on'],
```

Comment:
as per slack channel, too complicated to do on this PR. And not even sure it's worth it atm. For PA we create those dates so there is no risk of failure, and Lay deputies would need a clear message if interval is wrong (arleady done by the FE)